.github/workflows/tests_linux.yml

```diff
@@ -132,6 +132,66 @@ jobs:
           files: ./BuildCov/coverage.info
           fail_ci_if_error: true
 
+  cpptestswithkokkos:
+    name: C++ tests (Linux, Kokkos and Kokkos Kernels)
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.7'
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get -y -q install cmake gcc g++ gcovr lcov
+
+      - name: Build and run unit tests
+        run: |
+            cmake . -BBuild  -DCMAKE_BUILD_TYPE=Release -DENABLE_KOKKOS=ON -DBUILD_TESTS=1
+            cmake --build ./Build
+            mkdir -p ./Build/tests/results
+            ./Build/pennylane_lightning/src/tests/runner --order lex --reporter junit --out ./Build/tests/results/report.xml
+
+      - name: Upload test results
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: ubuntu-test-report
+          path: Build/tests/results/report.xml
+
+      - name: Publish test results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          check_name: Test Report (C++) on Ubuntu
+          files: Build/tests/results/report.xml
+
+      - name: Build and run unit tests for code coverage
+        run: |
+            cmake pennylane_lightning/src -BBuildCov  -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON -DENABLE_COVERAGE=ON -DENABLE_KOKKOS=ON
+            cmake --build ./BuildCov
+            cd ./BuildCov
+            ./tests/runner
+            lcov --directory . -b ../pennylane_lightning/src --capture --output-file coverage.info
+            lcov --remove coverage.info '/usr/*' --output-file coverage.info
```

Comment:
In hindight, we should be able to do this as part of the previous test, rather than compile and run the suite twice. Also, why do we need the debug flag here?

If it is too hard to combine the two, then maybe creating this as a separate test suite to run in parallel is a good idea.